Add support for reading TMN ESWC replays

kugelrund opened the pull request:
Depends on this change https://github.com/donadigo/pygbx/pull/11 in pygbx.

kugelrund committed:
Fix backup parse function not being used

kugelrund committed:
Add support for TMN ESWC replays

donadigo commented on average_steer_partitions.py:
```diff
@@ -21,6 +21,18 @@ def event_to_analog_value(event: ControlEntry):
     val >>= int32(8)
     return -val
 
+def clean_duplicate_events(events: list):
```
Please annotate that this also returns list.

kugelrund replied:
done

donadigo commented on average_steer_partitions.py:
```diff
@@ -21,6 +21,18 @@ def event_to_analog_value(event: ControlEntry):
+    """
+    events_cleaned = []
+    for i in range(0, (len(events)-1)):
+        if int(events[i+1].time / 10) * 10 != int(events[i].time / 10) * 10:
```
What happens when there's events like this?
```
10000 Accelerate (1)
10000 Brake (1)
10010 Accelerate (0)
```
Is the first accelerate simply skipped?

kugelrund replied:
Very good point. Reworked the function to skip over everything that is not a `Steer` event

donadigo commented on generate_input_file.py:
```diff
@@ -54,6 +54,18 @@ def event_to_analog_value(event: ControlEntry):
     val >>= int32(8)
     return -val
 
+def clean_duplicate_events(events: list):
```
Same here.

kugelrund replied:
done

donadigo reviewed (changes requested):
Some comments for now.

kugelrund committed:
Fix cleaning of duplicate steer events for ESWC

- Only clean steer events and leave other events completely untouched
- round up time to next 10 instead of midway rounding
